### ISSUE TITLE
import_raw() for BrainVision without electrode locations present

### ISSUE THREAD

**Issue by alexenge** [craddm/eegUtils#67]
Dear Matt,

Thanks for your great work on eegUtils! I'm very glad that you've added support for importing BrainVision files. However, when I try to import my own recordings, I get the following error message:

```r
raw <- eegUtils::import_raw("/Users/alexander/Research/Aha/Aha1/Results/EEG/raw/Vp0001.vhdr")
#> Importing Brain Vision Analyzer file /Users/alexander/Research/Aha/Aha1/Results/EEG/raw/Vp0001.vhdr
#> Error in names(new_coords) <- c("radius", "theta", "phi"): 'names' attribute [3] must be the same length as the vector [0]
```

Presumably because my `.vhdr` files don't contain any channel locations. Would it be possible to modify the `import_raw()` function to work without importing the locations (and, say, add some standard locations via `electrode_locations()` later on)?

Thanks & best
Alexander

**Comment by craddm** [comment]
Hi Alex,

Thanks! That rings a bell - I think I may have fixed that in the develop branch already. Try that, if you're not on it already, but do let me know if you are on that branch and still having problems.

Cheers,
Matt

**Comment by alexenge** [comment]
Thanks, Matt, for pointing me to the developer branch! Indeed, this seems to solve the issue with the missing electrode locations. However, I still get an error when trying to load my BrainVision files:

```r
raw <- import_raw(file_name = "Vp0001.vhdr")
#> Importing Brain Vision Analyzer file Vp0001.vhdr
#> No channel locations found.
#> Warning in (function (..., deparse.level = 1) : number of columns of result is
#> not a multiple of vector length (arg 1)
#> Warning: The `x` argument of `as_tibble.matrix()` must have column names if `.name_repair` is omitted as of tibble 2.0.0.
#> Using compatibility `.name_repair`.
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_warnings()` to see where this warning was generated.
#> Error: Can't bind data because all arguments must be named
rlang::last_error()
#> <error/rlang_error>
#> Can't bind data because all arguments must be named
#> Backtrace:
#>   1. eegUtils::import_raw(file_name = "Vp0001.vhdr")
#>   2. eegUtils:::import_vhdr(file_name, recording = recording, participant_id = participant_id)
#>   3. eegUtils:::read_vhdr(file_name)
#>   4. eegUtils:::read_vmrk(vmrk_file)
#>   6. dplyr:::mutate.data.frame(...)
#>   7. dplyr:::mutate_cols(.data, ...)
#>   8. DataMask$new(.data, caller_env())
#>   9. .subset2(public_bind_env, "initialize")(...)
#>  12. rlang::env_bind_lazy(...)
#>  13. rlang:::env_bind_impl(.env, exprs, "env_bind_lazy()", TRUE, binder)
#> Run `rlang::last_trace()` to see the full context.
rlang::last_trace()
#> <error/rlang_error>
#> Can't bind data because all arguments must be named
#> Backtrace:
#>      █
#>   1. └─eegUtils::import_raw(file_name = "Vp0001.vhdr")
#>   2.   └─eegUtils:::import_vhdr(file_name, recording = recording, participant_id = participant_id)
#>   3.     └─eegUtils:::read_vhdr(file_name)
#>   4.       └─eegUtils:::read_vmrk(vmrk_file)
#>   5.         ├─dplyr::mutate(...)
#>   6.         └─dplyr:::mutate.data.frame(...)
#>   7.           └─dplyr:::mutate_cols(.data, ...)
#>   8.             └─DataMask$new(.data, caller_env())
#>   9.               └─.subset2(public_bind_env, "initialize")(...)
#>  10.                 ├─base::suppressWarnings(...)
#>  11.                 │ └─base::withCallingHandlers(...)
#>  12.                 └─rlang::env_bind_lazy(...)
#>  13.                   └─rlang:::env_bind_impl(.env, exprs, "env_bind_lazy()", TRUE, binder)
```

System info:

```r
sessionInfo()
#> R version 4.0.2 (2020-06-22)
#> Platform: x86_64-apple-darwin17.0 (64-bit)
#> Running under: macOS Catalina 10.15.5
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_4.0.2  magrittr_1.5    tools_4.0.2     htmltools_0.5.0
#>  [5] yaml_2.2.1      stringi_1.4.6   rmarkdown_2.3   highr_0.8      
#>  [9] knitr_1.29      stringr_1.4.0   xfun_0.15       digest_0.6.25  
#> [13] rlang_0.4.6     evaluate_0.14
```

It would be awesome if you could take a look at this. I'm also happy to forward you one of my EEG data sets in case that would help. Thanks!

**Comment by craddm** [comment]
Hi @alexenge, sure, if you're happy to send some over that'd be great - I can't replicate it with the test data I have.

**Comment by craddm** [comment]
Error related to date field in markers, fixed by 3ac2c4e2c2a055b0d9876a06aeca6ad4f86c55cd
